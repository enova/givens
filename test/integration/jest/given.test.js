describe('basic overriding behavior', () => {
  given('var1', () => 'initial value');

  given('var2', () => `the value is: ${given.var1}`);

  it('has initial value', () => {
    expect(given.var2).toEqual('the value is: initial value');
  });

  describe('with new value', () => {
    given('var1', () => 'overridden value');

    it('has overridden value', () => {
      expect(given.var2).toEqual('the value is: overridden value');
    });
  });

  it('has initial value again', () => {
    expect(given.var2).toEqual('the value is: initial value');
  });
});
