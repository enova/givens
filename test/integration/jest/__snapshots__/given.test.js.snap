// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with recursive given variables blows up: recursive variables error 1`] = `
"givens: recursive variable recursive1 detected
trace: recursive1 => recursive2 => recursive1"
`;
